File: ios/Extopy/UI/Components/Users/UserCounterView.swift
//
//  UserCounterView.swift
//  Extopy
//
//  Created by Nathan Fallet on 13/01/2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import SwiftUI
import shared

struct UserCounterView: View {
    
    let type: UserCounter
    let value: Int64
    let onClick: () -> Void
    
    var body: some View {
        Button(action: onClick) {
            Text(stringForCounter.localized().format(value.simplify()))
        }
    }
    
    var stringForCounter: String {
        switch (type) {
        case .posts:
            return "timeline_info_posts"
        case .followers:
            return"timeline_info_followers"
        case .following:
            return "timeline_info_following"
        default:
            fatalError()
        }
    }
    
}

File: shared/src/commonMain/kotlin/com/extopy/ui/components/users/UserCounterView.kt
package com.extopy.ui.components.users

import androidx.compose.foundation.clickable
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.style.TextAlign
import com.extopy.extensions.simplify
import com.extopy.models.users.UserCounter
import extopy_app.shared.generated.resources.Res
import extopy_app.shared.generated.resources.timeline_info_followers
import extopy_app.shared.generated.resources.timeline_info_following
import extopy_app.shared.generated.resources.timeline_info_posts
import org.jetbrains.compose.resources.StringResource
import org.jetbrains.compose.resources.stringResource

@Composable
fun UserCounterView(
    type: UserCounter,
    value: Long,
    onClick: () -> Unit,
) {

    Text(
        text = stringResource(stringForCounter(type), value.simplify()),
        style = MaterialTheme.typography.bodyMedium,
        textAlign = TextAlign.Center,
        modifier = Modifier.clickable(onClick = onClick)
    )

}

fun stringForCounter(type: UserCounter): StringResource = when (type) {
    UserCounter.POSTS -> Res.string.timeline_info_posts
    UserCounter.FOLLOWERS -> Res.string.timeline_info_followers
    UserCounter.FOLLOWING -> Res.string.timeline_info_following
}
